### PR TITLE
Open modal from class

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description
This PR makes it possible to open a modal by a class, ex—a service, without inlining it. 

## References

- #981

### CODE REVIEW

File: libs/watt/src/lib/components/modal/watt-modal.service.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  /**
+   * Closes the modal with `true` for acceptance or `false` for rejection.
+   * @ignore
+   */
+  close(result: boolean) {
+    this.closeSubject.next(result);
+  }
+}
+
+@NgModule({
```
Comment (mimse) [inline libs/watt/src/lib/components/modal/watt-modal.service.ts:97]: Question: Module, why not standalone?

**Review by mimse — DISMISSED** [review]

File: libs/watt/src/lib/components/modal/watt-modal.service.ts (hunk 1 above)
Comment (frontend-specialisten) [reply libs/watt/src/lib/components/modal/watt-modal.service.ts:97]: I'm not a fan of it either, however, I was not able to make it work due to DI not having this module (at least this is only for internal usage).

File: libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    setTimeout(() => {
+      this.isLoading = false;
+    }, 300);
+  }
+}
+
+@Component({
+  standalone: true,
+  selector: 'watt-modal-from-class',
```
Comment (dzhavat) [inline libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts:55]: nitpick: the component has the same selector as the previous component.

File: libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { FormControl } from '@angular/forms';
+
+@Component({
+  standalone: true,
+  selector: 'watt-modal-from-class',
+  imports: [WATT_MODAL, WattTextFieldComponent, WattButtonComponent],
+  template: `
+    <watt-modal #modal [title]="title" closeLabel="Close modal" [loading]="isLoading">
+      <watt-text-field [formControl]="exampleFormControl" label="Username" />
+      <watt-text-field [formControl]="exampleFormControl" label="Password" type="password" />
```
Comment (dzhavat) [inline libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts:31]: nitpick: the same control is used on both input fields.

File: libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { WattTextFieldComponent } from '../../text-field/watt-text-field.component';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  standalone: true,
+  selector: 'watt-modal-from-class',
+  imports: [WATT_MODAL, WattTextFieldComponent, WattButtonComponent],
+  template: `
+    <watt-modal #modal [title]="title" closeLabel="Close modal" [loading]="isLoading">
+      <watt-text-field [formControl]="exampleFormControl" label="Username" />
```
Comment (dzhavat) [inline libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts:30]: issue: weird but when I type in the input, there's an error in the console.

image

File: libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts (hunk 2 above)
Comment (frontend-specialisten) [reply libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts:55]: It should be fixed now; the selector of the component used in the modal has now been removed.

File: libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts (hunk 3 above)
Comment (frontend-specialisten) [reply libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts]: Should be fixed now

File: libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts (hunk 4 above)
Comment (frontend-specialisten) [reply libs/watt/src/lib/components/modal/+storybook/watt-modal-from-class.component.ts]: The error was caused by not importing `ReactiveFormsModule` (good catch!).

**Review by dzhavat — APPROVED** [review]

**Pull request merged**